In [13]:
import os
import numpy as np
import pylab as pl
import json
from scipy.optimize import curve_fit,minimize
c=json.load(open('pui.json'))
pl.rcParams.update(c)

%pylab inline



Populating the interactive namespace from numpy and matplotlib


In [14]:
alpha = 0.05
P_0= 52.1*0.01
P_1=53.3*0.01
n_0=409
n_1=564
nt_0=P_0*n_0
nt_1=P_1*n_1


In [15]:
#define the sample proportion first
sp=(P_0*n_0+P_1*n_1)/(n_1+n_0)
print sp

0.527955806783


In [17]:
# i am goonna create a little one line function to calculate the standard dev, it is not really needed, but just to show you how you do such a thing
sp_stdev= lambda p, n: np.sqrt( p * ( 1 - p ) /n[0] +  p * ( 1 - p )/n[1]  )


sp_stdev_2y=sp_stdev((nt_0+nt_1)/(n_0+n_1),[n_0,n_1])
print P_0, n_0, n_1, sp_stdev_2y

0.521 409 564 0.0324224126866


In [18]:
zscore = lambda p0, p1, s : (p0-p1)/s
z_2y = zscore(P_1, P_0, sp_stdev_2y)
print z_2y


0.370114343926


In [20]:
p_2y=1-0.6368
def report_result(p,a):
    print 'is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a)
    if p<a:
        print "YES!"
    else: print "NO!"
    
    print 'the Null hypothesis is {}'.format( 'rejected' if p<a  else 'not rejected') 

    
report_result(p_2y,alpha)


The history saving thread hit an unexpected error (OperationalError('disk I/O error',)).History will not be written to the database.
is the p value 0.36 smaller than the critical value 0.05? 
NO!
the Null hypothesis is not rejected


In [21]:
Ntot = 973
expected = 564*409*131.579*841.421
sample_values = [[14.7*5.64,85.3*5.64],[11.9*4.09,88.1*4.09]]
 
chisqstat= lambda N, values, expect : N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expect)

print chisqstat(Ntot,  sample_values, expected)

1.58939333193


we have 1 degree of freedom and 0.05 is alpha so intersecting their respective row and column we get 3.84. 1.589 is close to 3.84. So we cannot reject the null hypothesis . AS the p-value for the corresponding value is not below 0.05. (can see from the table of 'Percentage points of Chi-Square distribution')